src/override-handler/adapter.ts

```diff
@@ -0,0 +1,66 @@
+import type { TApiRecorderData, THTTPStream } from './api-recorder.types';
+
+type AdapterContract = {
+  requestGroups(apiRecorderJson: TApiRecorderData[]): TRequestGroup[];
+};
+
+class OverrideHandlerAdapter implements AdapterContract {
+  requestGroups(apiRecorderJson: TApiRecorderData[]): TRequestGroup[] {
+    const groups = new Map<string, TApiRecorderData[]>();
+
+    for (const data of apiRecorderJson) {
+      if (data.type === 'http-rest' || data.type === 'http-stream') {
+        const key = `${data.request.method}:${data.request.url}`;
+
+        if (!groups.has(key)) {
+          groups.set(key, []);
+        }
+        const group = groups.get(key);
+        if (group) {
+          if (data.type === 'http-stream') {
+            group.push({
+              ...data,
+              parsedStreamEvents: this.#parseStreamData(data),
+            });
+          } else {
+            group.push(data);
+          }
+        }
+      }
+    }
```

Comment:
```suggestion
    for (const data of apiRecorderJson) {
      if (data.type !== 'http-rest' && data.type !== 'http-stream') {
        continue;
      }

      const key = `${data.request.method}:${data.request.url}`;
      const group = groups.get(key) || [];
      const finalData = match(data)
        .with({ type: 'http-stream' }, streamData => ({
          ...streamData,
          parsedStreamEvents: this.#parseStreamData(streamData),
        }))
        .with({ type: 'http-rest' }, restData => restData)
        .otherwise(() => null);

      if (finalData) {
        group.push(finalData);
        groups.set(key, group);
      }
    }
```

이렇게 바꾸면 어떨까요?
중첩 if문이 줄어서 가독성도 좋아지고, data.type은 시나리오 타입 파일 덕분에 잘못된 값이 들어올 가능성은 거의 없겠지만, if-else로 처리하는 것보다는 모든 경우를 패턴 매칭으로 명확히 구분하는 편이 더 좋을 것 같습니다!